clients/drcachesim/drptpostprocessor/drptpostprocessor.c

```diff
@@ -0,0 +1,649 @@
+                 * If so, change the image in pt_insn_deocer to the target image. */
+                image = NULL;
+                errcode = pt_sb_event(decoder->sb_session, &image, &event, sizeof(event),
+                                      stdout, 0);
+                if (errcode < 0)
+                    break;
+                if (image == NULL)
+                    continue;
+                /* pt_sb_event() use absolute path to load image. When we run this client
+                 * on a different machine than the trace recorded, the sideband session
+                 * may not be able to find the image. Then pt_sb_event() will return a not
+                 * NULL image, but the image is empty. So we need to add the required
+                 * image sections to preload the cache and copy them when the image switch
+                 * event happens.
+                 * XXX: This is a temporary solution. But it has one bug
+                 * for some applications that load images dynamically. If two images share
+                 * the same address, we can not decide which one is the target image. The
+                 * best solution is to use the relative path to remap the image. But this
+                 * needs to modify the code of the libipt sideband.
+                 */
+                pt_image_copy(image, decoder->preload_image);
+
+                errcode = pt_insn_set_image(decoder->instr_decoder, image);
+                if (errcode < 0) {
+                    break;
+                }
+            }
+            if (nextstatus < 0) {
+                diagnose_error(decoder->instr_decoder, nextstatus,
+                               "handle insn event error", insn.ip);
+                break;
+            }
+            if (errcode < 0) {
+                diagnose_error(decoder->instr_decoder, errcode,
+                               "handle sideband event error", insn.ip);
+                break;
+            }
+
+            if ((nextstatus & pts_eos) != 0) {
+                break;
+            }
+
+            /* Decode instructions. */
+            status = pt_insn_next(decoder->instr_decoder, &insn, sizeof(insn));
+            if (status < 0) {
+                diagnose_error(decoder->instr_decoder, status, "decode error", insn.ip);
+                break;
+            }
+
+            output->instr_count += 1;
+            /* TODO i#5505: Converting insn into an instruction-only memref_t */
+        }
+    }
+}
+
+static void
+print_stats(dript_output_t *output IN)
+{
+    printf("Number of Instructions: %" PRIu64 ".\n", output->instr_count);
+}
+
+static int
+usage(const char *prog IN)
+{
+    printf("Usage: %s [<options>]", prog);
+    printf("Command-line tool for decoding a PT trace, and converting it into an "
+           "instruction-only memtrace composed of 'memref_t's.\n");
+    printf("This version only counts and prints the instruction count in the trace "
```

Comment:
We can skip this printf. And modify the one above to say, "Command-line tool that decodes the given PT trace and returns the outputs as specified by given flags"

Reply:
Done.